python/dolfinx/fem/petsc.py

```diff
@@ -112,18 +110,48 @@ def create_vector(
 ) -> PETSc.Vec:
     """Create a PETSc vector that is compatible with a linear form(s).
 
-    If the vector type is not specified (``kind=None``) or is
-    ``PETSc.Vec.Type.MPI``, a ghosted PETSc vector which is compatible
-    with ``L`` is created. If the vector type is
-    ``PETSc.Vec.Type.NEST``, a PETSc nested vector (a nest of ghosted
-    PETSc vectors) which is compatible with ``L`` is created.
+    Three cases are supported:
+
+    1. For a single linear form ``L``, if ``kind`` is ``None`` or is
+       ``PETSc.Vec.Type.MPI``, a ghosted PETSc vector which is
+       compatible with ``L`` is created.
+
+    2. For a sequence of linear forms ``L``, if ``kind`` is ``None`` or
+       is ``PETSc.Vec.Type.MPI``, a ghosted PETSc vector which is
+       compatible with ``L`` is created. The created vector ``b`` is
+       initialized such that on each MPI process ``b = [b_0, b_1, ...,
+       b_n, b_0g, b_1g, ..., b_ng]``, where ``b_i`` are the entries
+       associated with the 'owned' degrees-of-freedom for ``L[i]`` and
+       ``b_ig`` are the 'unowned' (ghost) entries for ``L[i]``.
+
+       For this case, the returned vector has an attribute ``_blocks``
+       that holds the local offsets into ``b`` for the (i) owned and
+       (ii) ghost entries for each ``L[i]``. It can be accessed by
+       ``b.getAttr("_blocks")``. The offsets can be used to get views
+       into ``b`` for blocks, e.g.::
+
+           >>> offsets0, offsets1, = b.getAttr("_blocks")
+           >>> offsets0
+           (0, 12, 28)
+           >>> offsets1
+           (28, 32, 35)
+           >>> b0_owned = b.array[offsets0[0]:offsets0[1]]
+           >>> b0_ghost = b.array[offsets1[0]:offsets1[1]]
+           >>> b1_owned = b.array[offsets0[1]:offsets0[2]]
+           >>> b1_ghost = b.array[offsets1[1]:offsets1[2]]
+
+
+    3. If ``kind`` is ``PETSc.Vec.Type.NEST``, a PETSc nested vector (a
```

Comment:
Should this also say whether L is a sequence of linear forms, a single linear form, or both?